From 14c3f59f5157885ad8f941f0bad6c0c5e3db12f8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <michel.daenzer@amd.com>
Date: Mon, 21 Nov 2016 18:45:30 +0900
Subject: [PATCH 10/11] Call ValidateGC after ChangeClip in
 radeon_sync_scanout_pixmaps

The wrong order meant that the clipping region wasn't actually applied,
so it always copied the full contents from the other scanout pixmap.

Reviewed-by: Alex Deucher <alexander.deucher@amd.com>
---
 src/radeon_kms.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/radeon_kms.c b/src/radeon_kms.c
index 8bb261e2..5764c20c 100644
--- a/src/radeon_kms.c
+++ b/src/radeon_kms.c
@@ -495,8 +495,8 @@ radeon_sync_scanout_pixmaps(xf86CrtcPtr xf86_crtc, RegionPtr new_region,
 
     gc = GetScratchGC(dst->depth, pScreen);
     if (gc) {
-	ValidateGC(dst, gc);
 	gc->funcs->ChangeClip(gc, CT_REGION, sync_region, 0);
+	ValidateGC(dst, gc);
 	sync_region = NULL;
 	gc->ops->CopyArea(src, dst, gc, 0, 0, dst->width, dst->height, 0, 0);
 	FreeScratchGC(gc);
-- 
2.11.0

